include/ck/tensor_operation/gpu/device/device_softmax.hpp

```diff
@@ -170,21 +204,24 @@ struct DeviceSoftmax : public BaseOperator
     std::unique_ptr<BaseArgument> MakeArgumentPointer(const std::vector<index_t> inLengths,
                                                       const std::vector<index_t> inStrides,
                                                       const std::vector<int> reduceDims,
-                                                      AccDataType alpha,
-                                                      AccDataType beta,
+                                                      void* alpha,
+                                                      void* beta,
                                                       const void* in_dev,
-                                                      void* out_dev)
+                                                      void* out_dev) override
     {
         return std::make_unique<Argument>(inLengths,
                                           inStrides,
                                           reduceDims,
-                                          alpha,
-                                          beta,
+                                          *static_cast<AccDataType*>(alpha),
+                                          *static_cast<AccDataType*>(beta),
```

Comment:
We probably need to document the specification used here.  Reduction will align with this. 

And we might need more discussion, since:

1. The cuDNN and MIOpen specification for `scaling parameter` is more complicated than just using `AccDataType as type of alpha, beta`
2. For some Reduction operation, like MAX/MIN/AMAX `AccDataType` is usually the same as `InOutDataType` for best accessing performance.  In that case, whether it is most reasonable to use `AccDataType` for alpha/beta when `AccDataType` is half_t or int8_t
3. And another problem is, whether we really need to use `alpha` and `beta` for scaling and bias when Reduction is `MAX/MIN/AMAX` ?

Reply:
> The cuDNN and MIOpen specification for scaling parameter is more complicated than just using AccDataType as type of alpha, beta

What would it be? I couldn't find a lot concrete info on [MIOpen doc](https://rocmsoftwareplatform.github.io/MIOpen/doc/html/softmax.html#). And cuDNN [documents about scaling precision type](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#scaling-parameters) but not the accumulation type, either.

> For some Reduction operation, like MAX/MIN/AMAX AccDataType is usually the same as InOutDataType for best accessing performance. In that case, whether it is most reasonable to use AccDataType for alpha/beta when AccDataType is half_t or int8_t
> And another problem is, whether we really need to use alpha and beta for scaling and bias when Reduction is MAX/MIN/AMAX 

True that there's no point promoting to higher precision when it's max/min reduction. I can make max reduction part of softmax normalization leverage it. The last part of your question about reduction alpha/beta scaling is confusing to me. Is the question for reduction itself or softmax normalization?

Reply:
> > The cuDNN and MIOpen specification for scaling parameter is more complicated than just using AccDataType as type of alpha, beta
> 
> What would it be? I couldn't find a lot concrete info on [MIOpen doc]

(https://rocmsoftwareplatform.github.io/MIOpen/doc/html/softmax.html#). And cuDNN [documents about scaling precision type](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#scaling-parameters) but not the accumulation type, either.
> 
> > For some Reduction operation, like MAX/MIN/AMAX AccDataType is usually the same as InOutDataType for best accessing performance. In that case, whether it is most reasonable to use AccDataType for alpha/beta when AccDataType is half_t or int8_t
> > And another problem is, whether we really need to use alpha and beta for scaling and bias when Reduction is MAX/MIN/AMAX
> 
> True that there's no point promoting to higher precision when it's max/min reduction. I can make max reduction part of softmax normalization leverage it. The last part of your question about reduction alpha/beta scaling is confusing to me. Is the question for reduction itself or softmax normalization?

I think before  3) is clarified,  use `AccDataType *` is most reasonable.  Since: 
>  When in case of  Reduction MAX/MIN/AMAX,  we can assume alpha and beta will not be actually used, so it does not matter whether to use `AccDataType` or higher precision type for alpha/beta
>  In case of Reduction `Add/Mul/NORM1/NORM2`,  `AccDataType` is already required to be of higher precision, so it is not a problem to use `AccDataType` for alpha, beta

Reply:
So the question is mostly about reduction itself, not normalization? What actions do I need to take here?